--- /home/vagrant/FreeCAD/src/Mod/Part/App/PropertyTopoShape.cpp	2020-06-30 13:26:29.496897228 +0000
+++ /home/vagrant/FreeCAD/src/Mod/Part/App/PropertyTopoShape.cpp.new	2020-06-30 15:03:23.364419225 +0000
@@ -281,9 +281,9 @@
 static Standard_Boolean  BRepTools_Write(const TopoDS_Shape& Sh, 
                                    const Standard_CString File)
 {
-  ofstream os;
+  std::ofstream os;
 #if OCC_VERSION_HEX >= 0x060800
-  OSD_OpenStream(os, File, ios::out);
+  OSD_OpenStream(os, File, std::ios::out);
 #else
   os.open(File, ios::out);
 #endif
